pkg/vdri/bloc/config/consortium_test.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package config
+
+import (
+	"encoding/base64"
+	"testing"
+
+	"github.com/square/go-jose"
+	"github.com/stretchr/testify/require"
+)
+
+// nolint: gochecknoglobals
+var payload = `
+{
+	"domain": "foo.bar",
+	"policy": {
+		"cache": {"max-age": 123456789}
+	},
+	"stakeholders": [
+		{
+			"domain": "bar.baz",
+			"did": "did:bloc:foo.bar:zQ1234567890987654321"
+		},
+		{
+			"domain": "baz.qux",
+			"did": "did:bloc:foo.bar:zQ0987654321234567890"
+		}
+	],
+	"previous": ""
+}
+`
+
+// nolint: gochecknoglobals
+var exampleStakeholders = `[
+{
+	"domain": "bar.baz",
+	"did": "did:bloc:foo.bar:zQ1234567890987654321",
+	"conf": {
+		"cache": {"max-age": 123456789}
+	},
+	"endpoints": [
+		"https://bar.baz/webapi/123456",
+		"https://bar.baz/webapi/654321"
+	],
+	"previous": "testTest12345"
+},
+{
+	"domain": "baz.qux",
+	"did": "did:bloc:foo.bar:zQ0987654321234567890",
+	"conf": {
+		"cache": {"max-age": 123456789}
+	},
+	"endpoints": [
+		"https://baz.qux/iyoubhlkn/",
+		"https://baz.foo/ukjhjtfyw/"
+	],
+	"previous": "testTest67890"
+}]
+`
+
+func Test_ParseConsortium(t *testing.T) {
+	payloadB64 := base64.RawURLEncoding.EncodeToString([]byte(payload))
+
+	jws := `{"payload":"` + payloadB64 + `","signatures":[{"header":{"kid":""}, "signature":""}]}`
+
+	_, err := jose.ParseSigned(jws)
+	require.NoError(t, err)
+
+	fileData := jws + "\n" + exampleStakeholders
```

Comment:
Our plan is to store these files in DCAS and our DCAS implementation currently can only support a single JSON object in a document. Can we store the stakeholders as a field in the first doc? Otherwise we need to have the stakeholders live in a separate doc.

Reply:
go-jose doesn't play nicely with JWS's stored within another JSON object - it can parse a JWS that's in JSON string representation, but can't parse a JWS from an input stream, meaning I can't (easily) nest the JWS within another JSON object (for which I could use go's json.Decoder). Do you know a way around that?

Reply:
Our DCAS will eventually need to be modified to support non-JSON docs (so we can store things like images). Even so, I think it looks kind of weird to be looking for a newline delimiter within a file??

Reply:
I agree, it _is_ weird, and I don't like it either - I want to find a way around the delimeter. Right now I can see a few ways:
- Requires golang 1.14: use [json.Decoder.InputOffset](https://golang.org/pkg/encoding/json/#Decoder.InputOffset), letting me find the portion of the serialized string which holds only the JWS, so go-jose can parse that substring.
- JSON stringify the JWS so it can be stored as a string inside the document.
- Parse the JWS into a map while parsing the document, re-serialize the JWS into a string that go-jose can parse and process.
- Modify our custom implementation of JWS so it can parse a JSON JWS from an input stream (using json.Decoder) instead of needing the exact string.

The last option seems cleanest.

Reply:
Do we actually need the stakeholders to be in the same file? This list can be derived from the Consortium config, can't it?

Reply:
Troy wanted them cached in the same file to reduce the number of GETs needed for endpoint discovery. As a simple solution we can skip that caching, require the discovery service to fetch individual stakeholder config documents, and get the endpoints that way.

Reply:
I wouldn't use such a delimiter. We should return properly formatted JSON.

If we need to create a combined cache file then we can make that is own file (but as properly formatted JSON).

Reply:
Even without a delimeter, with the JWS as an element of a properly-formatted JSON structure, I think we'd need one of the options I mention above, so we can parse and process the JWS (using go-jose, subject to its API limitations, or extending it ourselves).

Reply:
The PR now amends the spec to have each document contain a separate JSON object, and implements the parser as such.